update_ui.py

```diff
@@ -0,0 +1,71 @@
+from subprocess import Popen, PIPE
+from ui.utility import find_ui_update
+from ui.config_loader import load
+import os
+import logging
+from datetime import datetime
+import shutil
+import tarfile
+
+'''
+UI updater
+'''
+
+# setup the logs
+log_format = logging.Formatter(
+        '%(asctime)s - %(name)-16s - %(levelname)-8s - %(message)s')
+log = logging.getLogger('')
+log.setLevel(logging.DEBUG)
+ch = logging.StreamHandler()
+ch.setFormatter(log_format)
+log.addHandler(ch)
+fh = logging.FileHandler('update.log')
+fh.setFormatter(log_format)
+log.addHandler(fh)
+
+'''
+checks the state, if state is set to in_progress, get the update file and continue
+'''
+log.info("checking state")
+process = Popen(["python", "ui/initial_state.py"], stdout=PIPE)
```

Comment:
because the only programs that can sucessfully read the pickled state file have to run from within the ui directory. As we are concerned with moving a directory that is the CWD of a running program, there has to be a way of checking state from outside the directory. 
So initial_state.py when run, prints out the update_ui state and that's read by the update_ui.py program.